### PR TITLE
[1304] Run geocoding job after receving an individual course

### PR DESCRIPTION
**Pull request by tvararu**
### Context

We need to set geolocations on a newly published location immediately after publishing, to prevent users from having up to 2 minutes of them not showing up on the map.

### Changes proposed in this pull request

Moves all the Geocoder modules inside of the Api domain, and then runs them after publishing one course.

### Guidance to review

... I am not proud of this code. :D

### CODE REVIEW

**Comment by tvararu** [comment]
[Spike: geocoding changed training locations immediately after publication [2 hours]](https://trello.com/c/EujJbLPy/1304-spike-geocoding-changed-training-locations-immediately-after-publication)

**Comment by misaka** [comment]
Looks reasonable to me, with my limited understanding of C#.

File: src/api/Services/Geocoder/LocationRequester.cs [hunk 1]
```diff
@@ -61,16 +61,23 @@ public async Task<int> RequestLocations()
...
+                    }
+                    else
+                    {
+                        failures.Add(location.Id.ToString(), location.GeoAddress);
+                    }
                 }
-                else
+                catch (Exception)
                 {
                     failures.Add(location.Id.ToString(), location.GeoAddress);
```
Comment (timabell) [inline src/api/Services/Geocoder/LocationRequester.cs]: can we include the exception info somewhere useful? maybe log to insights as a warning

Comment (defong) [reply src/api/Services/Geocoder/LocationRequester.cs]: Good call, forgot about it.

Logged as error as if there is exception it is quite terminal.

**Comment by defong** [comment]
Actioned

**Pull request merged**